src/event/mod.rs

```diff
@@ -136,6 +136,25 @@ impl std::ops::Index<&Atom> for LogEvent {
     }
 }
 
+// Allow converting any kind of appropriate key/value iterator directly into a LogEvent.
+impl<K: Into<Atom>, V: Into<ValueKind>, T: Iterator<Item = (K, V)>> From<T> for LogEvent {
```

Comment:
I think we probably want to implement `FromIterator` instead?

Reply:
Of course. My bad.